packages/mu-http/driver.js

```diff
@@ -14,4 +14,144 @@
 
 'use strict'
 
-// TODO
+var http = require('http')
+var stringify = require('fast-safe-stringify')
+var parse = require('fast-json-parse')
+var assert = require('assert')
+var mue = require('mu-error')()
+
+
+/**
+ * HTTP transport
+ * Figure out path mapping - perhaps this may need to be part of the config for now just /
+ */
+module.exports = function createTcpDriver (options) {
+  return function httpDriver (opts, receive) {
+    var connections = {}
+    var connectionsByIp = {}
+    var server
+
+    assert(opts, 'transport should always pass opts to httpDriver')
+    assert(receive instanceof Function, 'transport should always pass receive function to httpDriver')
+
+
+    function send (message, cb) {
+      var sendTo
+
+      if (connections[message.protocol.dst]) {
+
+        // server responding to client initaiated connection
+        if (connections[message.protocol.dst].length > 0) {
+          sendTo = connections[message.protocol.dst].pop()
+          sendTo.response.writeHead(200, { 'Content-Type': 'application/json' })
+          sendTo.response.write(stringify(message))
+          sendTo.response.end()
+        } else {
+          cb && cb(mue.transport('connection has alredy recieved response!'))
+        }
+      } else {
+
+        // client initiating connection to server
+        var body = []
+        var inbound
+        var req = http.request({host: options.target.host,
+                                port: options.target.port,
+                                method: 'POST',
+                                path: '/',
+                                headers: {'Content-Type': 'application/json'}})
```

Comment:
You should use an `Agent` here to speed things up. creating a new TCP connection for each things.